### ISSUE TITLE
Lazy-loaded module in new project gives "ɵɵdefineComponent is not a function" / Fixed by changing .browserslistrc

### ISSUE THREAD

**Issue by davehasagithub** [angular/angular#38820]
# 🐞 bug report

### Affected Package

 Unknown

### Is this a regression?

 Possibly. A new project created in version 9.1 does not have this issue.

### Description

I created a new project today using the following:

    ng new testapp --packageManager=yarn --skipGit=true --routing=true --style=scss
    ng generate module test --module=app --routing --route=test
    (then I emptied app.component.html and added <router-outlet></router-outlet>)
    ng build

When I visit the /test route in the _latest_ Firefox or Chrome, the developer console shows the error below. Running from the webpack dev server does not exhibit the issue.

This is strange, but I determined that the problem goes away after editing .browserslistrc to change "last 1 Chrome version" to "last 25 Chrome versions". Today this equates to Chrome versions 60-85. Using 'last 1' thru 24 all fail. It's also fixed by adding "> 0.5%". This existed in a new install with Angular 9. It seems this equates to just Chrome versions 83-85.

Sorry ahead of time. I'm aware this could be all my fault or maybe a bug in the browserslist library. Just seems like a brand new install would be fine and it took a long time to track down. I checked similar lazy-load examples in the tutorials and they work, but the Stackblitz projects I reviewed didn't have a browserslist file.

## 🔬 Minimal Reproduction

https://github.com/davehasagithub/ngtemp

First commit was from ng new, second from ng generate plus an updated app.component.html--that's it!

## 🔥 Exception or Error
<pre><code>ERROR Error: Uncaught (in promise): TypeError: _angular_core__WEBPACK_IMPORTED_MODULE_0__.ɵɵdefineComponent is not a function
TypeError: _angular_core__WEBPACK_IMPORTED_MODULE_0__.ɵɵdefineComponent is not a function
    at Module.K9yR (test.component.ts:8)
    at __webpack_require__ (bootstrap:84)
    at Module.NBQd (test-routing.module.ts:1)
    at __webpack_require__ (bootstrap:84)
    at Module.FScs (test.module.ts:1)
    at __webpack_require__ (bootstrap:84)
    at ZoneDelegate.invoke (zone-evergreen.js:364)
    at Object.onInvoke (core.js:27431)
    at ZoneDelegate.invoke (zone-evergreen.js:363)
    at Zone.run (zone-evergreen.js:123)
    at resolvePromise (zone-evergreen.js:798)
    at resolvePromise (zone-evergreen.js:750)
    at zone-evergreen.js:860
    at ZoneDelegate.invokeTask (zone-evergreen.js:399)
    at Object.onInvokeTask (core.js:27419)
    at ZoneDelegate.invokeTask (zone-evergreen.js:398)
    at Zone.runTask (zone-evergreen.js:167)
    at drainMicroTaskQueue (zone-evergreen.js:569)
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>Angular CLI: 10.1.1
Node: 14.10.0
OS: linux x64

Angular: 10.1.1
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.1001.1
@angular-devkit/build-angular     0.1001.1
@angular-devkit/build-optimizer   0.1001.1
@angular-devkit/build-webpack     0.1001.1
@angular-devkit/core              10.1.1
@angular-devkit/schematics        10.1.1
@ngtools/webpack                  10.1.1
@schematics/angular               10.1.1
@schematics/update                0.1001.1
rxjs                              6.6.3
typescript                        4.0.2
webpack                           4.44.1</code></pre>

**Anything else relevant?**

FWIW, I'm running the CLI in a linux VirtualBox in a synced folder and the browser in the Windows host.

**Comment by JoostK** [comment]
Interestingly, I cannot reproduce using your exact repo (on macOS, though). I'm running `yarn ng serve` to run the app, then navigate to `http://localhost:4200/test` and that works as expected, without any errors.

Could you remove all of `node_modules`, then reinstall and once that has completed, run your app again and share the exact command you use and all its output here? Maybe that sheds some light on what's going on, as currently I don't really have an idea of what could be going on here.

**Comment by davehasagithub** [comment]
Hi, thanks for the reply. The issue doesn't occur with the dev server. I only noticed after using ng build, and serving with apache.

I set the dist folder as the docroot and add an [.htaccess](https://github.com/davehasagithub/ngtemp/blob/master/_misc/.htaccess) to redirect paths to index.html. I don't think that's involved though since I see the same error when clicking a routerLink to /test.

I've recreated node_modules many times. Maybe it'll be useful if I generate the dist folder both ways and compare the two.

**Comment by JoostK** [comment]
I just tested your latest changes using `ng build` and Python's SimpleHTTPServer (`python -m SimpleHTTPServer 4200`) as that's the simplest way I can think of to run the app. That's working fine for me as well. I don't know about Apache as I don't have that setup. If you can add a docker-compose setup to the repo for me to run it using Apache then I'd be happy to see what's going on.

**Comment by davehasagithub** [comment]
I think this will help. I pasted this into my ssh window:

    rm -rf node_modules dist dist_bad dist_good
    yarn install
    ng build
    mv dist dist_bad
    sed -i "s/last 1 Chrome version/last 25 Chrome versions/g" .browserslistrc
    ng build
    mv dist dist_good

The results are very different: [bad](https://github.com/davehasagithub/ngtemp/tree/master/dist_bad/testapp) and [good](https://github.com/davehasagithub/ngtemp/tree/master/dist_good/testapp). Just comparing index.html files, you can see runtime.js vs runtime-es2015.js. I also noticed the first ng build says "Compiling @angular/core : es2015 as esm2015" and the second says "Generating ES5 bundles for differential loading". But I don't understand why.

**Comment by JoostK** [comment]
Yeah so that is the difference between having differential loading disabled vs enabled. Using latest 25 Chrome versions infers that it needs to generate ES5 bundles, so that's what it does. A recent Chrome browser however should still only load the ES2015 bundle, so there shouldn't be a difference there.

Still, it doesn't explain the issue you're seeing. The ES2015 files should be identical to the files generated if differential loading is disabled. I can spot one difference, though; if differential loading is disabled, the ES2015 script include in index.html will not have `type="module"`. Maybe you could experiment with adding that in in the situation where it fails (again, I can not reproduce the error you're seeing).

**Comment by davehasagithub** [comment]
You're right! I changed `defer` to `type=module` in all 5 script tags of the failing index.html and now it works in both chrome and firefox. Does that mean my setup is generating it incorrectly? Or could it be a mac vs windows browser issue?

    <script src="runtime.js" defer></script>
    <script src="polyfills.js" defer></script>
    <script src="styles.js" defer></script>
    <script src="vendor.js" defer></script>
    <script src="main.js" defer></script>

    <script src="runtime.js" type="module"></script>
    <script src="polyfills.js" type="module"></script>
    <script src="styles.js" type="module"></script>
    <script src="vendor.js" type="module"></script>
    <script src="main.js" type="module"></script>

**Comment by JoostK** [comment]
I don't know, actually. I'll transfer to the CLI repo as that's where differential loading + index.html processing is implemented.
